src/python/pants/backend/python/packaging/pyoxidizer/rules.py

```diff
@@ -51,91 +52,80 @@ class PyOxidizerFieldSet(PackageFieldSet):
 async def package_pyoxidizer_binary(
     pyoxidizer: PyOxidizer, field_set: PyOxidizerFieldSet
 ) -> BuiltPackage:
-    logger.debug(f"Incoming package_pyoxidizer_binary field set: {field_set}")
-    targets = await Get(Targets, DependenciesRequest(field_set.dependencies))
-    target = targets[0]
-
-    logger.debug(f"Received these targets inside pyox targets: {target.address.target_name}")
-
-    packages = await Get(
-        FieldSetsPerTarget,
-        FieldSetsPerTargetRequest(PackageFieldSet, [target]),
+    direct_deps, pyoxidizer_pex = await MultiGet(
+        Get(Targets, DependenciesRequest(field_set.dependencies)),
+        Get(
+            Pex,
+            PexRequest(
+                output_filename="pyoxidizer.pex",
+                internal_only=True,
+                requirements=pyoxidizer.pex_requirements(),
+                interpreter_constraints=pyoxidizer.interpreter_constraints,
+                main=pyoxidizer.main,
+            ),
+        ),
     )
-    logger.debug(f"Retrieved the following FieldSetsPerTarget {packages}")
 
+    deps_field_sets = await Get(
+        FieldSetsPerTarget, FieldSetsPerTargetRequest(PackageFieldSet, [direct_deps[0]])
+    )
     built_packages = await MultiGet(
-        Get(BuiltPackage, PackageFieldSet, field_set) for field_set in packages.field_sets
+        Get(BuiltPackage, PackageFieldSet, field_set) for field_set in deps_field_sets.field_sets
     )
-
-    wheels = [
+    wheel_paths = [
```

Comment:
Just need to clarify my understanding on this:
The built_packages will be either a wheel, or an sdist (depending on the target settings), and then we're just filtering out the wheel.

What happens if we end up with an empty array? I think I meant to have a TODO here for an explicit assertion or warning/error - because otherwise, we try pip installing an empty array inside the config if memory serves?

Reply:
I'm not sure what happens yet, but I want to investigate it tomorrow and fix via:

> Figure out the story for multiple python_distributions, especially if you depend on one that depends on another one.

Part of that is the story for _no_ valid python_distributions.

Right now `dependencies` assumes exactly 1 valid `python_distribution` target. We want to explicitly handle 0 and >1.

Reply:
Okay, good, same boat - I'll be pulling it down tomorrow to mess around. PyOxidizer **should** bail out, but that's from messing around a while back

Reply:
Cool, thanks! FYI I'm planning on working on the `output_path` ticket first tomorrow. We definitely don't need things to be perfect for PyOxidizer, but it'd be good to make big API changes like that before 2.10 stabilizes